kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/StatsSchemaHelper.java

```diff
@@ -0,0 +1,304 @@
+     *  |    |-- numRecords: long (nullable = false)
+     *  |    |-- minValues: struct (nullable = false)
+     *  |    |    |-- a: struct (nullable = false)
+     *  |    |    |    |-- b: struct (nullable = false)
+     *  |    |    |    |    |-- c: long (nullable = true)
+     *  |    |-- maxValues: struct (nullable = false)
+     *  |    |    |-- a: struct (nullable = false)
+     *  |    |    |    |-- b: struct (nullable = false)
+     *  |    |    |    |    |-- c: long (nullable = true)
+     *  |    |-- nullCount: struct (nullable = false)
+     *  |    |    |-- a: struct (nullable = false)
+     *  |    |    |    |-- b: struct (nullable = false)
+     *  |    |    |    |    |-- c: long (nullable = true)
+     *  }}}
+     */
+    public static StructType getStatsSchema(StructType dataSchema) {
+        StructType statsSchema = new StructType()
+            .add(NUM_RECORDS, LongType.LONG, true);
+        StructType minMaxStatsSchema = getMinMaxStatsSchema(dataSchema);
+        if (minMaxStatsSchema.length() > 0) {
+            statsSchema = statsSchema
+                .add(MIN, getMinMaxStatsSchema(dataSchema), true)
+                .add(MAX, getMinMaxStatsSchema(dataSchema), true);
+        }
+        StructType nullCountSchema = getNullCountSchema(dataSchema);
+        if (nullCountSchema.length() > 0) {
+            statsSchema = statsSchema
+                .add(NULL_COUNT, getNullCountSchema(dataSchema), true);
+        }
+        return statsSchema;
+    }
+
+    //////////////////////////////////////////////////////////////////////////////////
+    // Instance fields and public methods
+    //////////////////////////////////////////////////////////////////////////////////
+
+    private final StructType dataSchema;
+    /* Map of all leaf columns from logical to physical names */
+    private final Map<Column, Column> logicalToPhysicalColumn;
+    /* Map of all leaf logical columns to their data type */
+    private final Map<Column, DataType> logicalToDataType;
+
+    public StatsSchemaHelper(StructType dataSchema) {
+        this.dataSchema = dataSchema;
+        Map<Column, Tuple2<Column, DataType>> logicalToPhysicalColumnAndDataType =
+            getLogicalToPhysicalColumnAndDataType(dataSchema);
+        this.logicalToPhysicalColumn = logicalToPhysicalColumnAndDataType.entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    Map.Entry::getKey,
+                    e -> e.getValue()._1 // map to just the column
+                )
+            );
+        this.logicalToDataType =
+            logicalToPhysicalColumnAndDataType.entrySet().stream()
+                .collect(
+                    Collectors.toMap(
+                        Map.Entry::getKey,
+                        e -> e.getValue()._2 // map to just the data type
+                    )
+                );
+    }
+
+    /** Given a logical column returns corresponding the MIN column in the statistic schema */
```

Comment:
`Given a logical column returns corresponding the MIN column` I don't get this sentence btw.
`returns corresponding` is an odd phrasing

Reply:
Updated this let me know if it's more clear; otherwise I can try to be even more specific